src/cpp/_pythia8.cpp

```diff
@@ -103,6 +104,28 @@ py::array_t<int> event_array_children(Event &event)
     return py::array_t<int>(shape, strides, ptr1);
 }
 
+// refills "event" stack with particles 
+// having pid and energy
+void refill_stack(Event &event,
```

Comment:
Please change name to `fill_event` When you fill something, it is usually implied that the original content is deleted.

Reply:
Actually no, the name should be just `fill`, since this is a method of `Event`. Repeating "event" in the name is redundant.

Reply:
I change the name to `fill` and put it as method for Event.